hl_extractor/hl_calc.py

```diff
@@ -62,21 +64,27 @@ def _calculate(self):
             subprocess.check_call([os.path.join(".", HIGH_LEVEL_EXTRACTOR_BINARY),
                                    name, out_file, PROFILE_CONF],
                                   stdout=fnull, stderr=fnull)
-        except subprocess.CalledProcessError:
+        except (subprocess.CalledProcessError, OSError):
             print("Cannot call high-level extractor")
+            # If we return early, make sure we remove the temp
+            # output file that we created
+            os.unlink(out_file)
             return "{}"
-
-        fnull.close()
-        os.unlink(name)
+        finally:
+            # At this point we can remove the source file,
+            # regardless of if we failed or if we succeeded
+            fnull.close()
+            os.unlink(name)
 
         try:
             f = open(out_file)
             hl_data = f.read()
             f.close()
-            os.unlink(out_file)
         except IOError:
             print("IO Error while removing temp file")
```

Comment:
This `IOError` no longer gets thrown on *removing* the temp file, just when reading it.